### PR TITLE
pkg/morph/container: Do not create client with notary status

### PR DESCRIPTION
**Pull request by carpawell**
Provide notary signature status with explicit parameter. Some calls are always expected to be Alphabet signed (at least for the current contracts), make it explicit for a code reader, that is more error proof. This commit also fixes container size estimations writing: they were tried to be notary signed, but the Alphabet does not handle such requests. This is also the reason why this kinda refactor does not touch other contracts; general client improvements can be done with another PR.

### CODE REVIEW

File: pkg/morph/client/static.go [hunk 1]
```diff
@@ -97,6 +99,14 @@ func (i *InvokePrmOptional) SetHash(hash util.Uint256) {
 	i.hash = &hash
 }
 
+// NotaryAlphabetSignature makes client send notary request instead of a
+// regular signed transaction. Such a request should be received and signed by
+// the Alphabet, otherwise the plug (empty) transaction will be added to the
+// chain.
+func (ti *InvokePrmOptional) NotaryAlphabetSignature(signByAlphabet bool) {
```
Comment (roman-khimov) [inline pkg/morph/client/static.go:106]: Can be `RequireAlphabetSignature()`, the default is `false`, this will just set it to `true`.

File: pkg/morph/client/container/client.go [hunk 2]
```diff
@@ -90,26 +86,13 @@ type opts struct {
...
 func defaultOpts() *opts {
 	return new(opts)
 }
 
-// DisableNotarySigning returns option to disable
-// notary request signing. With that option, every
-// call will be created, signed with provided key
-// (a single regular sign) and sent to the side
-// chain.
-func DisableNotarySigning() Option {
```
Comment (roman-khimov) [inline pkg/morph/client/container/client.go:107]: This doesn't seem to be used, can be a separate patch.

**Review by roman-khimov — COMMENTED** [review]
While the intention is good, I fear it can break things.

**Review by roman-khimov — COMMENTED** [review]
How about `balance`? Likely you can't mint/burn without alphabet support. NeoFSID? Netmap?

**Comment by roman-khimov** [comment]
You only change `container` defaults though, so likely these are fine.

File: pkg/morph/client/container/client.go (hunk 2 above)
Comment (carpawell) [reply pkg/morph/client/container/client.go:107]: ok

File: pkg/morph/client/static.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/static.go]: dont mind, wanted it to be flexible, but that is just prm struct, not needed. fixed

File: pkg/morph/client/container/client.go [hunk 3]
```diff
@@ -60,10 +60,6 @@ func NewFromMorph(cli *client.Client, contract util.Uint160, fee fixedn.Fixed8,
 		o.staticOpts = append(o.staticOpts, client.WithCustomFee(putNamedMethod, o.feePutNamed))
 	}
 
-	if !o.disableNotarySigning {
-		o.staticOpts = append(o.staticOpts, client.TryNotary())
```
Comment (roman-khimov) [inline pkg/morph/client/container/client.go:64]: But this changes the default behavior, so it's not just `drop ...`

Comment (carpawell) [reply pkg/morph/client/container/client.go:64]: ok, chagned the message, check it out

Comment (carpawell) [reply pkg/morph/client/container/client.go:64]: changed one more time. let defaults be changed in the first commit while the second can be still "Drop"

**Comment by roman-khimov** [comment]
Yup, convincing.

**Review by roman-khimov — APPROVED** [review]
